Speed up sound source list by using an unordered map

Automat-GH opened the pull request:
This avoids iterating the entire list whenever a new sound source is created (which was contributing to the stuttering mentioned in #142 and #131).

Commit:
Speed up sound source list by using an unordered map
This avoids iterating the entire list whenever a new sound source is created (which was contributing to the stuttering mentioned in #142 and #131).

Xottab-DUTY commented on src/xrSound/SoundRender_Core.h:
```diff
@@ -54,7 +55,7 @@ class CSoundRender_Core : public ISoundManager
 
     // Containers
-    xr_vector<CSoundRender_Source*> s_sources;
+    xr_unordered_map<std::string, CSoundRender_Source*> s_sources;
```
Please, don't use `std::` types, since we have our own. At least we should use `xr_string`... But, can you try to use `shared_str` here?

Xottab-DUTY commented on src/xrSound/SoundRender_Core.cpp:
```diff
@@ -86,8 +86,10 @@ void CSoundRender_Core::_clear()
+    for (auto it = s_sources.begin(); it != s_sources.end(); ++it)
+    {
+        xr_delete(it->second);
+    }
```
We can use range-based for here

Xottab-DUTY commented on src/xrSound/SoundRender_Core.cpp:
```diff
@@ -592,9 +594,9 @@ void CSoundRender_Core::refresh_sources()
     for (u32 eit = 0; eit < s_emitters.size(); eit++)
         s_emitters[eit]->stop(false);
-    for (u32 sit = 0; sit < s_sources.size(); sit++)
+    for (auto it = s_sources.begin(); it != s_sources.end(); ++it)
```
We can use range-based for here

Commit:
Changed std::string to xr_string
Use range based for instead of manual iterators
Auto for find iterator instead of explicit type

Commit:
put xr_delete back
